src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -19,28 +20,60 @@
 
     private String mergeRemote;
     private String mergeTarget;
-    private String mergeStrategy;
+    private MergeCommand.Strategy mergeStrategy;
     private MergeCommand.GitPluginFastForwardMode fastForwardMode;
 
     /**
      * @deprecated use the new constructor that allows to set the fast forward mode.
      */
     @Deprecated
     public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy) {
-        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
+        this(mergeTarget);
+        setMergeRemote(mergeRemote);
+        setMergeStrategy(mergeStrategy);
+        setFastForwardMode(MergeCommand.GitPluginFastForwardMode.FF);
+    }
+
+    /**
+     * @Deprecated use the new @DataBoundConstructor where you only need to supply the necessary information.
+     */
+    @Deprecated
+    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy, MergeCommand.GitPluginFastForwardMode fastForwardMode) {
+        this(mergeTarget);
+        setMergeRemote(mergeRemote);
+        setMergeStrategy(mergeStrategy);
+        setFastForwardMode(fastForwardMode);
     }
 
     @DataBoundConstructor
-    public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy,
-            MergeCommand.GitPluginFastForwardMode fastForwardMode) {
-        this.mergeRemote = mergeRemote;
+    public UserMergeOptions(String mergeTarget) {
         this.mergeTarget = mergeTarget;
+    }
+
+    @DataBoundSetter
+    public void setMergeRemote(String mergeRemote) {
+        this.mergeRemote = mergeRemote;
+    }
+
+    @DataBoundSetter
+    public void setMergeStrategy(MergeCommand.Strategy mergeStrategy) {
         this.mergeStrategy = mergeStrategy;
+    }
+
+    @DataBoundSetter
+    public void setFastForwardMode(MergeCommand.GitPluginFastForwardMode fastForwardMode) {
         this.fastForwardMode = fastForwardMode;
     }
 
+    private void setMergeStrategy(String mergeStrategy) {
+        this.mergeStrategy = (mergeStrategy == null ? null : MergeCommand.Strategy.valueOf(mergeStrategy.toUpperCase()));
```

Comment:
Why the `.toUpperCase()`? I see no indication that case variants were accepted before.